Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1476,51 +1482,152 @@ extension BrowserViewController: WKUIDelegate {
+        })
+    }
+
+    func disableDialog(switchState: UISwitch) {
+        tabManager.selectedTab?.disableDialog = switchState.on
+    }
+
+    func removeDelegate() {
+    }
+
     func webView(webView: WKWebView, runJavaScriptAlertPanelWithMessage message: String, initiatedByFrame frame: WKFrameInfo, completionHandler: () -> Void) {
         tabManager.selectTab(tabManager[webView])
+        if let tab = tabManager.selectedTab {
+            var message = "\(message)"
+
+            if tab.showDialog {
+                // Allow two extra spaces to show disable Dialog prompt
+                message = "\(message)\n\n"
+            }
+
+            // Show JavaScript alerts.
+            let title = frame.request.URL!.host
+            let alertController = UIAlertController(title: title, message: message, preferredStyle: UIAlertControllerStyle.Alert)
+            alertController.addAction(UIAlertAction(title: OKString, style: UIAlertActionStyle.Default, handler: { _ in
+                completionHandler()
+                if tab.disableDialog {
+                    webView.UIDelegate = nil
+                }
+            }))
 
-        // Show JavaScript alerts.
-        let title = frame.request.URL!.host
-        let alertController = UIAlertController(title: title, message: message, preferredStyle: UIAlertControllerStyle.Alert)
-        alertController.addAction(UIAlertAction(title: OKString, style: UIAlertActionStyle.Default, handler: { _ in
-            completionHandler()
-        }))
-        presentViewController(alertController, animated: true, completion: nil)
+            if tab.showDialog {
+                createDisableDialogPrompt(alertController)
+            }
+
+            presentViewController(alertController, animated: true, completion: nil)
+            tab.showDialog = true
+        }
     }
 
     func webView(webView: WKWebView, runJavaScriptConfirmPanelWithMessage message: String, initiatedByFrame frame: WKFrameInfo, completionHandler: (Bool) -> Void) {
         tabManager.selectTab(tabManager[webView])
+        if let tab = tabManager.selectedTab {
+            var message = "\(message)"
 
-        // Show JavaScript confirm dialogs.
-        let title = frame.request.URL!.host
-        let alertController = UIAlertController(title: title, message: message, preferredStyle: UIAlertControllerStyle.Alert)
-        alertController.addAction(UIAlertAction(title: OKString, style: UIAlertActionStyle.Default, handler: { _ in
-            completionHandler(true)
-        }))
-        alertController.addAction(UIAlertAction(title: CancelString, style: UIAlertActionStyle.Cancel, handler: { _ in
-            completionHandler(false)
-        }))
-        presentViewController(alertController, animated: true, completion: nil)
+            if tab.showDialog {
+                // Allow two extra spaces to show disable Dialog prompt
+                message = "\(message)\n\n"
```

Comment:
Would really be nice if we could use layout for this instead of string whitespace.